sematic/ui/src/components/ReactFlowDag.tsx

```diff
@@ -66,6 +67,19 @@ function ReactFlowDag() {
     [edges]
   );
 
+  const onSelectArtifact = useCallback((node: Node) => {
+    const runId = node.data.sourceRunId || node.data.destinationRunId;
+    const artifactRun = graph?.runsById.get(runId);
+
+    if (artifactRun) {
+      setSelectedRunTab(node.data.sourceRunId ? "output" : "input");
+      //Labels do not exists for output artifacts and they are labelled as null.
+      setSelectedArtifactId(node.data.label || "null");
```

Comment:
Shall we use `node.id` here?

Reply:
I do not think so that works because `node.id` contains the run name not the artifact name.

Reply:
Thanks for mentioning that. 

I see, so `node.id` is 
```
let makeArtifactNodeId = (edge: Edge) =>
      (edge.source_run_id || "null") +
      (edge.destination_name || "null") +
      (edge.destination_run_id || "null");
```
which is an artificial UI ID.

Unfortunately, despite `edge` carries `artifact_id`, we do not have it in the `node.data`:

https://github.com/sematic-ai/sematic/blob/7d49e207f0dabf91d3cae1de805c35e692067807/sematic/ui/src/pipelines/graph/ReactFlowDag.tsx#L108-L120

I think what I would do is, in `node.data` 
```
node_data.push({
          type: "artifactNode",
          id: artifactNodeId,
          data: {
            label: edge.destination_name,
            nodeId: artifactNodeId,
            artifactId: edge.artifact_id, // note this new line
            sourceRunId: edge.source_run_id,
            destinationRunId: edge.destination_run_id,
          }, //getEdgeLabel(edge) },
          parentNode: parentId,
          position: { x: 0, y: 0 },
          zIndex: 0,
        });
```

Also due to this line

https://github.com/sematic-ai/sematic/blob/7d49e207f0dabf91d3cae1de805c35e692067807/sematic/ui/src/pipelines/RunDetailsPanel.tsx#L45

The downstream misuses the label as the key. 

Since we have `artifact_id`, I think we should use that as the unique identifier instead of using labels (though logically, parameter names won't conflict with each other, I guess relying on this coincidence should not be preferred.)

Please let me know if this makes sense and whether you would like to incorporate this change in this PR.

Reply:
I kind of get what you are referring to here. But even if we create an `artifactId` and reference it to scroll down instead of the artifact name. But then what do we do here? 

https://github.com/sematic-ai/sematic/blob/main/sematic/ui/src/pipelines/Artifacts.tsx#L70

The artifacts map contains name -> Artifact mapping.

Reply:
Yeah, this part needs to be clarified, indeed. And there was an oversight in my comment above as well. 

The code you cited does not need to be changed. However 

https://github.com/kamalesh0406/sematic/blob/270c723b3eff37464d285de534e92b6c82d7be67/sematic/ui/src/pipelines/Artifacts.tsx#L66

It can be changed to 
```typescript
  const artifactIds = useMemo(()=>Array.from(artifacts.values().map(artifact => artifact.id)), [artifacts]);
```

will that work?

Reply:
Yup that should work, let me test it out now.